crates/bevy_ecs/src/query/state.rs

```diff
@@ -158,6 +158,10 @@ impl<Q: WorldQuery, F: ReadOnlyWorldQuery> QueryState<Q, F> {
         }
     }
 
+    /// Panics if `world` does not match the one used to call `QueryState::new` for this instance.
+    ///
+    /// Many unsafe query methods require the world to match for soundness. This function is the easiest
+    /// way of ensuring that it matches.
```

Comment:
the "standard" way is to put panics in their own sections, see https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html for an example

```suggestion
    /// Many unsafe query methods require the world to match for soundness. This function is the easiest
    /// way of ensuring that it matches.
    ///
    /// # Panics
    ///
    /// Panics if `world` does not match the one used to call `QueryState::new` for this instance.
```